src/vehicle.h

```diff
@@ -858,6 +858,23 @@ class vehicle : public JsonSerializer, public JsonDeserializer
 
     /*@}*/
 
+    /**
+     * If target for the turret is set, it returns the set target and unsets target.
+     * Otherwise finds a target and returns it.
+     * @return A target for the turret or turret's position if no target is found.
+     */
+    tripoint turret_acquire_target( vehicle_part &pt );
+    /**
+     * Fires a loaded turret at the target.
+     * @return Number of shots fired.
+     */
+    int turret_fire( vehicle_part &pt, const tripoint &targ );
+
+    void turret_reload( vehicle_part &pt );
```

Comment:
>> These are an implementation detail
> Or even better: you can add a test of this type to your branch yourself, to make sure everything works.

Agreed with Coolthulu, when do you expect this branch to land?  If it's not "today" we should go ahead and land this PR (assuming everything works, haven't checked that).

Either way, make sure your new version has an API we can test.